Objects/typeobject.c

```diff
@@ -409,6 +486,21 @@ PyType_Modified(PyTypeObject *type)
         }
     }
 
+    if (type->tp_watched) {
+        PyInterpreterState *interp = _PyInterpreterState_GET();
+        char bits = type->tp_watched;
+        for (int i = 0; i < TYPE_MAX_WATCHERS; i++) {
```

Comment:
Given that the low bits are likely the only ones set, this could be sped up a bit by turning this into `while (bits)` (and making bits an `int`)